processor/src/lib.rs

```diff
@@ -0,0 +1,194 @@
+                                              self.registers.clone(),self.flag, None);
+                    self.pc += 1;
+                }
+                "lt" => {
+                    debug!("opcode: lt");
+                    assert!(ops.len() == 3 ,  "lt params len is 2");
+                    let dst_index = self.get_reg_index(ops.get(1).unwrap());
+                    // let src_index = self.get_reg_index(ops.get(2).unwrap());
+                    let value = self.get_index_value(ops.get(2).unwrap());
+                    self.flag = self.registers[dst_index] < value.0;
+                    program.trace.insert_step(self.clk, self.pc as u32, self.fp[self.fp.len()-1] as u32, Instruction::LT(Lt{ri: dst_index as u8, a: value.1}),
+                                              self.registers.clone(),self.flag, None);
+                    self.pc += 1;
+                }
+                "cjmp" => {
+                    debug!("opcode: cjmp");
+                    assert!(ops.len() == 2 ,  "cjmp params len is 1");
+                    let value = self.get_index_value(ops.get(1).unwrap());
+                    if self.flag == true {
+                        // fixme: use flag need reset?
+                        self.flag = false;
+                        program.trace.insert_step(self.clk, self.pc as u32, self.fp[self.fp.len()-1] as u32, Instruction::CJMP(CJmp{a: value.1}),
+                                                  self.registers.clone(),self.flag, None);
+                        self.fp.push(self.pc +1);
+                        self.pc = value.0;
+                    } else {
+                        program.trace.insert_step(self.clk, self.pc as u32, self.fp[self.fp.len()-1] as u32, Instruction::CJMP(CJmp{a: value.1}),
+                                                  self.registers.clone(),self.flag, None);
+                        self.pc += 1;
+                    }
+
+                }
+                "jmp" => {
+                    debug!("opcode: jmp");
+                    assert!(ops.len() == 2 ,  "jmp params len is 1");
+                    let value = self.get_index_value(ops.get(1).unwrap());
+                    program.trace.insert_step(self.clk, self.pc as u32, self.fp[self.fp.len()-1] as u32, Instruction::JMP(Jmp{a: value.1}),
+                                              self.registers.clone(),self.flag, None);
+                    self.fp.push(self.pc +1);
+                    self.pc = value.0;
+
+                }
+                "add" => {
+                    debug!("opcode: add");
+                    assert!(ops.len() == 4 ,  "add params len is 3");
+                    let dst_index = self.get_reg_index(ops.get(1).unwrap());
+                    let op1_index = self.get_reg_index(ops.get(2).unwrap());
+                    let op2_value = self.get_index_value(ops.get(3).unwrap());
+                    self.registers[dst_index] =  self.registers[op1_index] + op2_value.0;
```

Comment:
Need more code here: We can't do two array addition, and we also need consider overflow flag.